e2e/suite_test.go

```diff
@@ -0,0 +1,106 @@
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	daemonsetDesiredNumber = 1
+)
+
+func TestE2E(t *testing.T) {
+	if !runE2E {
+		t.Skip("no RUN_E2E environment variable")
+	}
+
+	RegisterFailHandler(Fail)
+	SetDefaultEventuallyTimeout(1 * time.Minute)
+	SetDefaultEventuallyPollingInterval(1 * time.Second)
+	RunSpecs(t, "E2E Suite")
+}
+
+var _ = BeforeSuite(func() {
+	By("prepare pod for profiling")
+	_, err := kubectl(nil, "apply", "-f", "./manifests/profiled-pod.yaml")
+	Expect(err).NotTo(HaveOccurred())
+
+	By("waiting for profiled-pod to be ready")
+	Eventually(func() error {
+		res, err := kubectl(nil, "get", "pod", "profiled-pod", "-o", "json")
+		if err != nil {
```

Comment:
This is just a matter of preference.
You can use Gomega's assertions even in `Eventually()`.

```
Eventually(func(g Gomega) {
  ...
  g.Expect(err).NotTo(HaveOccurred())
  ...
}).Should(Succeed())
```